File: src/test/java/com/github/redfoos/logstash/highlighting/HighlightingTest.java
package com.github.redfoos.logstash.highlighting;

import com.intellij.testFramework.fixtures.BasePlatformTestCase;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;

public class HighlightingTest extends BasePlatformTestCase {

    @Override
    protected String getTestDataPath() {
        return "src/test/testData/highlighting";
    }

    public void testPluginBlockDefinition() {
        doTest();
    }

    public void testFunctionalElement() {
        doTest();
    }

    private void doTest() {
        File file = new File(getTestDataPath(), getTestName(true) + ".conf");

        String content = "";
        try {
            content = new String(Files.readAllBytes(file.toPath()));
        } catch (IOException e) {
            e.printStackTrace();
        }

        myFixture.configureByText(file.getName(), content);
        myFixture.testHighlighting();

    }
}

File: src/test/kotlin/com/github/redfoos/logstash/hightlighting/HighlightingTest.kt
package com.github.redfoos.logstash.hightlighting

import com.intellij.testFramework.fixtures.BasePlatformTestCase
import org.junit.Test
import java.io.File
import java.io.IOException
import java.nio.file.Files

class HighlightingTest : BasePlatformTestCase() {
    override fun getTestDataPath(): String {
        return "src/test/testData/highlighting"
    }

    @Test
    fun testPluginBlockDefinition() {
        doTest()
    }

    @Test
    fun testFunctionalElement() {
        doTest()
    }

    private fun doTest() {
        val file = File(testDataPath, getTestName(true) + ".conf")
        var content = ""
        try {
            content = String(Files.readAllBytes(file.toPath()))
        } catch (e: IOException) {
            e.printStackTrace()
        }
        myFixture.configureByText(file.name, content)
        myFixture.testHighlighting()
    }
}
